### PR TITLE
Document setting up cloud provider kubectl auth via plugin (AKS, GKE)

### LINKED ISSUE #35896: Document setting up cloud provider kubectl auth via plugin (AKS, GKE)
**This is a Feature Request**

**What would you like to be added**
(As I understand it), Kubernetes ~v1.25~ v1.26 ships an updated `kubectl`. The updated `kubectl` won't include built-in plugins for authenticating to AKS or GKE.
This is likely to surprise folks.

Update 3 pages:
- https://kubernetes.io/docs/tasks/tools/install-kubectl-linux
- https://kubernetes.io/docs/tasks/tools/install-kubectl-macos
- https://kubernetes.io/docs/tasks/tools/install-kubectl-windows

to **signpost** users to the right internal ~and external~ docs. We can remove this signposting once Kubernetes v1.27 (sic) is the oldest supported version of Kubernetes (consider adding an HTML comment with that detail).

Optionally, also update https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins to mention that earlier versions of `kubectl` included built-in support for authenticating to AKS and GKE, but this is no longer present.

**Why is this needed**
Readers may follow the “install kubectl” docs and get a tool that's missing functionality they thought was built in.

**Comments**
> The gcp and azure auth plugins have been removed from client-go and kubectl. See [AKS](https://github.com/Azure/kubelogin) and [GKE](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke) documentation for details about the cloud-specific replacements.

/sig cli
/language en

### PR DESCRIPTION
**Pull request by coder12git**
Fixes #35896

### CODE REVIEW

File: content/en/docs/tasks/tools/included/verify-kubectl.md [hunk 1]
```diff
@@ -35,4 +35,15 @@ If kubectl cluster-info returns the url response but you can't access your clust
...
+{{< note >}}
+### Troubleshooting the 'No Auth Provider Found' Error Message
+
+In Kubernetes **v1.26**, kubectl removed the built-in authentication for the following cloud
+providers managed Kubernetes offerings. These providers have released kubectl plugins to provide the cloud-specific authentication. For instructions, refer to the following provider documentation:
+
+*  AKS (Azure): [kubelogin plugin](https://github.com/Azure/kubelogin)
+*  GKE (Google Cloud): [gke-gcloud-auth-plugin](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl#install_plugin)
+
+{{< /note >}}
```
Comment (sftim) [inline content/en/docs/tasks/tools/included/verify-kubectl.md:49]: We don't put headings inside callouts. Try this:
```suggestion
### Troubleshooting the 'No Auth Provider Found' error message {#no-auth-provider-found}

In Kubernetes 1.26, kubectl removed the built-in authentication for the following cloud
providers' managed Kubernetes offerings. These providers have released kubectl plugins to provide the cloud-specific authentication. For instructions, refer to the following provider documentation:

* Azure  AKS: [kubelogin plugin](https://github.com/Azure/kubelogin)
* Google Kubernetes Engine: [gke-gcloud-auth-plugin](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl#install_plugin)

(There could also be other reasons to see the same error message, unrelated to that change.)

```

**Review by sftim — COMMENTED** [review]
Thanks. I've added a suggestion.

**Comment by coder12git** [comment]
Done the changes!

File: content/en/docs/tasks/tools/included/verify-kubectl.md [hunk 2]
```diff
@@ -35,4 +35,14 @@ If kubectl cluster-info returns the url response but you can't access your clust
...
-```
\ No newline at end of file
+```
+
+### Troubleshooting the 'No Auth Provider Found' error message {#no-auth-provider-found}
+
+In Kubernetes 1.26, kubectl removed the built-in authentication for the following cloud
+providers' managed Kubernetes offerings. These providers have released kubectl plugins to provide the cloud-specific authentication. For instructions, refer to the following provider documentation:
+
+* Azure  AKS: [kubelogin plugin](https://github.com/Azure/kubelogin)
```
Comment (sftim) [inline content/en/docs/tasks/tools/included/verify-kubectl.md:45]: This could link to https://azure.github.io/kubelogin/ instead.

**Review by sftim — COMMENTED** [review]
Thanks

/lgtm

Because this change would go live as soon as we merge it, I'll let another approver add their approval.

**Pull request merged**